conda_build/build_index.py

```diff
@@ -0,0 +1,272 @@
+# Copyright (C) 2014 Anaconda, Inc
+# SPDX-License-Identifier: BSD-3-Clause
+"""
+conda-build's use of conda-index, delegated to now-separate conda-index package.
```

Comment:
Copied to make keeping track of all the changes easier, and to discourage future `conda_build.index` imports

Reply:
To clarify, we couldn't continue to have index.py as the place for this code to live? Or asked differently, the get_build_index function, is that something that we have in conda-index? I'm a little worried we're introducing another place where users will end up importing from :)

Reply:
`get_build_index` is only in conda-build. It is mostly about fetching repodata.json and not about generating repodata.json.

Reply:
However we might want to raise deprecationwarning when conda_build.index is imported.

Reply:
I don't think this submodule should exist.

Plus, all of `conda.index` should be deprecated with warnings to import from `conda-index` instead.

Reply:
Do you mean `conda.index` or `conda_build.index`

Reminder that conda-build's `get_build_index` doesn't belong in conda-index.

Reply:
@kenodegard @jezdez I moved this module back to conda_build.index